vispy/gloo/preprocessor.py

```diff
@@ -0,0 +1,79 @@
+
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright (c) 2015, Vispy Development Team. All Rights Reserved.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
+# -----------------------------------------------------------------------------
+import re
+from .. import glsl
+from ..util import logger
+
+
+def remove_comments(code):
+    """Remove C-style comment from GLSL code string."""
+
+    pattern = r"(\".*?\"|\'.*?\')|(/\*.*?\*/|//[^\r\n]*\n)"
+    # first group captures quoted strings (double or single)
+    # second group captures comments (//single-line or /* multi-line */)
+    regex = re.compile(pattern, re.MULTILINE | re.DOTALL)
+
+    def do_replace(match):
+        # if the 2nd group (capturing comments) is not None,
+        # it means we have captured a non-quoted (real) comment string.
+        if match.group(2) is not None:
+            return ""  # so we will return empty to remove the comment
+        else:  # otherwise, we will return the 1st group
+            return match.group(1)  # captured quoted-string
+
+    return regex.sub(do_replace, code)
+
+
+def remove_version(code):
+    """ Remove any version directive """
+
+    pattern = '\#\s*version[^\r\n]*\n'
+    regex = re.compile(pattern, re.MULTILINE | re.DOTALL)
+    return regex.sub('\n', code)
```

Comment:
This is done in gloo.Program now, which means that the user cannot specify the version explicitly anymore.